examples/timeseries/event_classification_for_payment_card_fraud_detection.py

```diff
@@ -0,0 +1,430 @@
+
+Future leakage is pernicious. Luckily, Temporian is here to help: The only operator that can cause future leakage is `tp.leak`. If you are not using `tp.leak`, your preprocessing is **guaranteed** not to contain future leakage.
+
+**Note:** For advanced pipelines, you can also check programatically that a feature does not depends on a `tp.leak` results.
+"""
+
+# Cast all values (e.g. ints) to floats.
+values = all_data[input_feature_names].cast(tp.float32)
+
+# Apply z-normalization overtime.
+normalized_features = (
+    values - values.simple_moving_average(math.inf)
+) / values.moving_standard_deviation(math.inf)
+
+# Restore the original name of the features.
+normalized_features = normalized_features.rename(values.schema.feature_names())
+
+print(normalized_features)
+
+"""
+The first transactions will be normalized using poor estimates of the mean and standard deviation since there are only a few transactions before them. To mitigate this issue, we remove the first week of data from the training dataset.
+
+Notice that the first values contain NaN. In Temporian, NaN represents missing values, and all operators handle them accordingly. For instance, when calculating a moving average, NaN values are not included in the calculation and do not generate a NaN result.
+
+However, neural networks cannot natively handle NaN values. So, we replace them with zeros.
+"""
+
+normalized_features = normalized_features.fillna(0.0)
+
+"""
+Finally, we group together the features and the labels.
+"""
+
+normalized_all_data = tp.glue(normalized_features, all_data["TX_FRAUD"])
+
+"""
+## Split dataset into a train, validation and test set
+
+To evaluate the quality of our machine learning model, we need training, validation and test sets. Since the system is dynamic (new fraud patterns are being created all the time), it is important for the training set to come before the validation set, and the validation set come before the testing set:
+
+- **Training:** April 8, 2018 to July 31, 2018
+- **Validation:** August 1, 2018 to August 31, 2018
+- **Testing:** September 1, 2018 to September 30, 2018
+
+For the example to run faster, we will effectively reduce the size of the training set to:
+- **Training:** July 1, 2018 to July 31, 2018
+"""
+
+# begin_train = datetime.datetime(2018, 4, 8).timestamp() # Full training dataset
+begin_train = datetime.datetime(
+    2018, 7, 1
+).timestamp()  # Reduced training dataset
+begin_valid = datetime.datetime(2018, 8, 1).timestamp()
+begin_test = datetime.datetime(2018, 9, 1).timestamp()
+
+is_train = (normalized_all_data.timestamps() >= begin_train) & (
+    normalized_all_data.timestamps() < begin_valid
+)
+is_valid = (normalized_all_data.timestamps() >= begin_valid) & (
+    normalized_all_data.timestamps() < begin_test
+)
+is_test = normalized_all_data.timestamps() >= begin_test
```

Comment:
I think the new `.before()` and `.after()` ops can handle this in a prettier way?

Reply:
I went back and forth on this. Doing the comparison manually (normalized_all_data.timestamps() >= begin_train) is more verbosy but has two benefits.

1) We need the intermediate results, not available in before/after to create the train/validation/test plot which is important to understand the time split.
2) The manual form is similar to pandas and therefore, possibly easier to understand than beginning.

What about keeping the manual form, and adding a comments explaining that the before/after operator is a more compact way to do this?